src/main/java/uk/gov/hmcts/cmc/claimstore/services/document/DocumentManagementBackedDocumentsService.java

```diff
@@ -86,7 +97,10 @@ public DocumentManagementBackedDocumentsService(
 
     @Override
     public byte[] generateDefendantResponseReceipt(String externalId, String authorisation) {
-        Claim claim = getClaimByExternalId(externalId, authorisation);
+        Claim claim = claimService.getClaimByExternalId(externalId, authorisation);
+        if (!claim.getResponse().isPresent() && null == claim.getRespondedAt()) {
+            throw new NotFoundException("Defendant response does not exist for this claim");
```

Comment:
Illegalstateexception than not found?

Reply:
ok will do

Reply:
keeping it as NotFoundException for now . We will need to revisit this .